AUTHORING_LIBRARIES.md

```diff
@@ -0,0 +1,58 @@
+
+There are a few reasons we need a build step:
+- use of pre-shipped JS features (decorators, and other in-progress ECMA proposals)
+- co-located components are not modules-by-default[^gjs]
+- building at publish time makes apps faster -- because the apps then have less to transform themselves[^embroider-compat-v2]
+
+If you'd like to provide build-time behaviors, using [unplugin](https://github.com/unjs/unplugin) allows you to write a build-time plugin once, using similar-to-rollup-APIs, and works for Vite, Rollup, Webpack, and ESBuild.
+
+## `package.json`
+
+For background on native packages ([V2 Addons](https://github.com/emberjs/rfcs/blob/master/text/0507-embroider-v2-package-format.md)) and what all is needed to publish real npm packages, this documentation will be helpful:
+- https://dev.to/binjospookie/exports-in-package-json-1fl2
+- https://nodejs.org/api/packages.html#package-entry-points
+- https://webpack.js.org/guides/package-exports/
+
+Using `package.json#exports` means that:
+- Used to specify what can be imported from your library. 
+- The `addon.publicEntrypoints(...)` plugin in `rollup.config.mjs` should _at least_, include whatever is in `package.json#exports`
+- The `addon.appReexports(...)` plugin must have overlap with the `package.json#exports` so that the app-tree merging may import from the addon.
+
+## Importing modules
+
+Modules should have the file extension included in the import path so that rollup knows which set of plugins to run over the file.
+
+```ts
+import { Something } from './path/to/file.ts';
+```
+
+If you've done ESM in node, this should feel familiar, and we can be be consistent with JS imports as well:
+
+```js
+import { AnotherThing } from './path/to/file.js';
+```
+
+A couple caveats with older, co-located components,
+ - for `.hbs` / template-only components, the import path is still `.js`, because we transform it.
+ - for co-located components, where the template is in a separate `.hbs` file, you may not import that `.hbs` file directly, because it is merged in with the associated `.js` or `.ts` file.
+
+
+## CSS
+
+The broader JavaScript ecosystem does not yet have a standardized way of handling CSS form libraries, but there are some things aligning. This [proposal-import-attributes](https://github.com/tc39/proposal-import-attributes) describes how packagers (such as Webpack, Vite, etc), can consistently, cross-ecosystem-ly, interpreted CSS imports and have CSS participate in the module graph. 
+
+We want CSS to participate in the module graph so that tree-shaking and bundle splitting can work for CSS as well as JS. Though, this applies to more than just CSS, and it will work for any file extension that you want to import. For example, importing SVG files could give you the SVG, the URL to the SVG, or a reference to a spritesheet.
+
+For now though, importing CSS in libraries requires some assumptions about what the consumers' build configuration is, and _may_ be packager-specific.
```

Comment:
maybe it's me, I think this section confuses more than helps because it may leave on without the answer to question "so how can I use css in my addon?"
if I'm not mistaken, we can leave a note about the need to import file with `.css` extension to get it bundled in the app?

Reply:
do you have suggestions about how to improve the spirit of the section?
The main thing is that: 

> a note about the need to import file with .css extension to get it bundled in the app?

this only works because of how we've set up webpack, it's totally possible that folks in the future may remove that capability from their apps.

Reply:
Thank you @NullVoxPopuli! Latest update looks great and IMO makes things clearer for folks who may not know some details.